policy/policy.go

```diff
@@ -293,7 +332,32 @@ type Resource struct {
 // level accept. match indicate a rule level match
 // and credential acceptance.
 func (r *Resource) Match(r1 *Resource) (bool, bool) {
-	accept, match := r.Name.Match(r1.Name)
+	// When matching an input resource against a resource defined by the policy, just consider
+	// the rules applied to the fields defined in the policy and ignore the rest
+	var fieldsToMatch []string
+	r.Name.foreach(func(rule *Rule) {
+		// TODO: We now hardcode the Pattern format but this would need to be reworked once we have
```

Comment:
I think even now we should be able to put this definition in the Rule struct.  At the very least it should be commented there, but I think an accessor that extracts the name and value parts can then be added to the RuleStruct.  (with the obvious waring about reading it directly :)